### ISSUE TITLE
Data attributes validation

### ISSUE THREAD

**Issue by 041616** [cheeriojs/cheerio#600]
When I'm trying to get data attributes, I have an error:
```js
var cheerio = require("cheerio");
var text = '<span data-some-info="{&#34;info&#34;: &#34;Important info here!&#34;}">title</span>'
var $ = cheerio.load(text, {decodeEntities: false});
$("*").each(function(){
  var dataAttrsObj = $(this).data();
})
```
```bash
undefined:1
{&#34;cn_model&#34;: {&#34;item_id&#34;: 1989, &#34;page_type&#34;: &#34;item_
 ^
SyntaxError: Unexpected token &
    at Object.parse (native)
```
inside data attributes I have something like this:
```html
<span
  data-some-info="{&#34;info&#34;: &#34;Important info here!&#34;}"
>title</span>
```
Why can't I put any string into data attribute?

**Comment by taramid** [comment]
Cheerio tries to parse data in case it thinks the data is a JSON and that decision is based on a naive RegEx test: `rbrace = /^(?:\{[\w\W]*\}|\[[\w\W]*\])$/;`

in `var readData = function(el, name) {`
there's:
      `} else if (rbrace.test(value)) {`
          `  value = JSON.parse(value);`
      `}`

IMHO Cheerios should NOT try to check/parse stored data as JSON (at least unless nobody asked)

because so far it fails too often. For example I try to scrap data with player nicknames and their comments. Half of those data look as a JSON data for Cheerio.

**Comment by fb55** [comment]
This was fixed by 4c0a1079f40aedd5d1abc38b548e49eb6b72c427.
